File: 프로그래머스/lv2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    var answer = [];
    let arr =progresses.map((e,i)=>Math.ceil((100-e)/speeds[i]));
    let m=0;
    let c=0;
    for(let i of arr){
        if(m==0){m = i;}
        else if(m<i){answer.push(c); c=0; m =i;}
        c++;
    }
    answer.push(c);
    return answer;
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
import math

def solution(progresses, speeds):
    answer = []
    task = []
    
    for i,v in enumerate(progresses):
        task.append(math.ceil((100 - v)/speeds[i]))  

    tmp = task[0]
    day = 0
    for x in task:
        if tmp >= x:
            day+=1
        else:
            answer.append(day)
            tmp = x
            day = 1
    answer.append(day)
    
    return answer